java/fury-core/src/main/java/org/apache/fury/serializer/collection/CollectionSerializer.java

```diff
@@ -45,6 +51,35 @@ public T onCollectionRead(Collection collection) {
     return (T) collection;
   }
 
+  @Override
+  public T copy(T originCollection) {
+    if (isImmutable()) {
+      return originCollection;
+    }
+    Collection newCollection;
+    if (!needToCopyRef) {
+      newCollection = newCollection(originCollection);
+      copyElements(originCollection, newCollection);
+      return (T) newCollection;
+    }
+    newCollection = (Collection) fury.getCopyObject(originCollection);
+    if (Objects.nonNull(newCollection)) {
+      return (T) newCollection;
+    }
+    fury.incCopyDepth(1);
+    newCollection = newCollection(originCollection);
+    fury.copyReference(originCollection, newCollection);
```

Comment:
`copyReference` seems just track ref for copy, how about name it as `reference(T o1, T o2)`, it's an overload method, if we give it a detailed javadoc, it won't introduce confusion